### PR TITLE
winrar.xml

### PR DESCRIPTION
**Pull request by Tobias-B-Besemer**

### CODE REVIEW

**Comment by Tobias-B-Besemer** [comment]
The WinRAR Cleaner in WinApp2.ini have this rule:
`FileKey1=%LocalAppData%\VirtualStore\Program Files*\WinRAR|*.tmp|RECURSE`
...and I'm not sure how BB handles this. The path have a Wildcard, searched are for *.tmp and it should be recurse... Can handle this BB at all in WinApp2.ini ???

I solved this now with the following two lines, but I don't know if it works:
`<action command="delete" search="walk.all" path="%LocalAppData%\VirtualStore\Program Files*\WinRAR\*.tmp"/>`
`<action command="delete" search="glob" path="%LocalAppData%\VirtualStore\Program Files*\WinRAR\*.tmp"/>`

**Comment by az0** [comment]
You asked about this line
```
FileKey1=%LocalAppData%\VirtualStore\Program Files*\WinRAR|*.tmp|RECURSE
````
Yes,  this should work. BleachBit converts it to CleanerML. If I remember correctly, this will become `search="walk.files"` with a regex filter.

**Comment by Tobias-B-Besemer** [comment]
OK, `search="walk.files"` can have a wildcard in the path, right?

But why a regex filter ???

Can you give me the code for my cleaner ???

File: cleaners/winrar.xml [hunk 1]
```diff
@@ -47,8 +49,8 @@
     <action command="winreg" path="HKCU\Software\WinRAR SFX"/>
   </option>
   <option id="temp">
-    <label>Temp Files</label>
-    <description>Deletes the Temp Files</description>
+    <label>Temporary files</label>
+    <description>Delete the temporary files</description>
```
Comment (az0) [inline cleaners/winrar.xml]: These two are translated. This is good. Tanks

File: cleaners/winrar.xml [hunk 2]
```diff
@@ -33,7 +35,7 @@
   <description>File archiver</description>
   <option id="private">
     <label>Private Data</label>
-    <description>Deletes the Private Data</description>
+    <description>Deletes the data</description>
```
Comment (az0) [inline cleaners/winrar.xml]: How about "Deletes the files"? It's already translated, and the label clarifies this is all about private data.

**Comment by Tobias-B-Besemer** [comment]
> You asked about this line
> 
> ```
> FileKey1=%LocalAppData%\VirtualStore\Program Files*\WinRAR|*.tmp|RECURSE
> ```
> 
> Yes, this should work. BleachBit converts it to CleanerML. If I remember correctly, this will become `search="walk.files"` with a regex filter.

I tested the following line in https://github.com/bleachbit/bleachbit/pull/462:
`<action command="delete" search="walk.all" path="$$profile$$\*.bak"/>`
...and this doesn't work, doesn't go recursive!
Any solution for this?

**Comment by Tobias-B-Besemer** [comment]
`<action command="delete" search="walk.files" path="$$profile$$\*.bak"/>`
...seems also not to solve the issue!

I want to delete those files:
Löschen 5.8kB C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azlocprov\GeoIP-Changes.bak
Löschen 26.4kB C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azlocprov\GeoIP-LICENSE.bak
Löschen 3.2kB C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azlocprov\GeoIP-README.bak
Löschen 194.3kB C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azneti2phelper\hosts.txt.bak
Löschen 245.9kB C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azneti2phelper\i2hostetag.b32.txt.bak
Löschen 342B C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azneti2phelper\README.txt.bak
Löschen 2kB C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azneti2phelper\certificates\reseed\backup_at_mail.i2p.crt.bak
Löschen 2kB C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azneti2phelper\certificates\reseed\bugme_at_mail.i2p.crt.bak
Löschen 2kB C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azneti2phelper\certificates\reseed\cheezybudz_at_mail.i2p.crt.bak
Löschen 2kB C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azneti2phelper\certificates\reseed\echelon_at_mail.i2p.crt.bak
Löschen 2.1kB C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azneti2phelper\certificates\reseed\j_at_torontocrypto.org.crt.bak
Löschen 2kB C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azneti2phelper\certificates\reseed\matt_at_drollette.com.crt.bak
Löschen 2kB C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azneti2phelper\certificates\reseed\meeh_at_mail.i2p.crt.bak
Löschen 2.1kB C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azneti2phelper\certificates\reseed\parg_at_mail.i2p.crt.bak
Löschen 2kB C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azneti2phelper\certificates\reseed\sindu_at_mail.i2p.crt.bak
Löschen 1.9kB C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azneti2phelper\certificates\reseed\swat_at_mail.i2p.crt.bak
Löschen 1.3kB C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azneti2phelper\certificates\ssl\193.150.121.66.crt.bak
Löschen 1.4kB C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azneti2phelper\certificates\ssl\i2p-netdb.innovatio.no.crt.bak
Löschen 1.4kB C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azneti2phelper\certificates\ssl\i2p.mooo.com.crt.bak
Löschen 1.4kB C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azneti2phelper\certificates\ssl\i2p.mooo.com2.crt.bak
Löschen 1.6kB C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azneti2phelper\certificates\ssl\i2pseed.zarrenspry.info.crt.bak
Löschen 1.5kB C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azneti2phelper\certificates\ssl\ieb9oopo.mooo.com.crt.bak
Löschen 1.5kB C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azneti2phelper\certificates\ssl\ieb9oopo.mooo.com2.crt.bak
Löschen 2kB C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azneti2phelper\certificates\ssl\jp.reseed.i2p2.no.crt.bak
Löschen 1.5kB C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azneti2phelper\certificates\ssl\link.mx24.eu.crt.bak
Löschen 1.4kB C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azneti2phelper\certificates\ssl\netdb.i2p2.no.crt.bak
Löschen 1.4kB C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azneti2phelper\certificates\ssl\netdb.i2p2.no2.crt.bak
Löschen 2kB C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azneti2phelper\certificates\ssl\netdb.rows.io.crt.bak
Löschen 1.4kB C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azneti2phelper\certificates\ssl\reseed.i2p-projekt.de.crt.bak
Löschen 2.1kB C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azneti2phelper\certificates\ssl\reseed.i2p.vzaws.com.crt.bak
Löschen 1.2kB C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azneti2phelper\certificates\ssl\reseed.info.crt.bak
Löschen 2kB C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azneti2phelper\certificates\ssl\uk.reseed.i2p2.no.crt.bak
Löschen 1.2kB C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azneti2phelper\certificates\ssl\us.reseed.i2p2.no.crt.bak
Löschen 1kB C:\Users\Administrator\AppData\Roaming\Azureus\plugins\azneti2phelper\certificates\ssl\user.mx24.eu.crt.bak

That all *.bak-files under the Roaming\Azureus-folder, I found with the Deep Scanner...

**Comment by az0** [comment]
You asked how to search for a recursive pattern. The page [CleanerML documentation](https://docs.bleachbit.org/doc/cleanerml.html) gives a little information about four regex attributes. One you may be familiar with from deep scan: `regex`. You can also use the same attribute with walk.files like this

`
<action command="delete" search="walk.files" path="%LocalAppData%\VirtualStore\Program Files*"\WinRAR" regex="*\.tmp" />
`
I didn't actualy test this one, so please do try it. Also, you may want to switch the path prefix to $$profile$$

There are some more examples of the regex attributes if you search the .xml files

**Comment by az0** [comment]
Here's a little tweak to the regex to make sure it matches the right side of the filename:

`regex="*\.tmp$"`

**Comment by Tobias-B-Besemer** [comment]
I guess it's better I use this: `regex="\.[Tt][Mm][Pp]$"` (?)

**Comment by Tobias-B-Besemer** [comment]
> Also, you may want to switch the path prefix to $$profile$$

As WinRAR is a Windows-only program, I don't see ATM a benefit in using $$profile$$.

File: cleaners/winrar.xml (hunk 1 above)
Comment (Tobias-B-Besemer) [reply cleaners/winrar.xml]: No prob! I get better... ;-)

File: cleaners/winrar.xml (hunk 2 above)
Comment (Tobias-B-Besemer) [reply cleaners/winrar.xml]: But no files get deleted... AFAIK I used now "Delete the data" and this in a couple of cleaners... It's really universal - more universal then "Delete the files" (also plural) - so I think it's really worth (also for the future, for other cleaners), to translate this string...

**Pull request merged**